File: API/Entities/Appointment.cs
using System;
using System.Text.Json.Serialization;

namespace API.Entities;

public class Appointment
{
    public int Id { get; set; }
    public DateOnly AppointmentDate { get; set; }
    public TimeOnly StartTime { get; set; }
    public TimeOnly EndTime { get; set; }
    public string? Notes { get; set; }  
    public AppointmentStatus Status { get; set; } = AppointmentStatus.Pending;
    public string ClientName { get; set; } = "";
    public string ClientEmail { get; set; } = "";
    public string? ClientPhone { get; set; }
    public bool ConfirmationNotificationSent { get; set; }
    public DateTime? ConfirmedAt { get; set; }
    public DateTime Created { get; set; } = DateTime.UtcNow;

    public string? ClientId { get; set; }
     [JsonIgnore]
    public AppUser? Client { get; set; }
    
    [JsonIgnore]
    public Review? Review { get; set; }
    

    [JsonIgnore]
    public ICollection<Service> Services { get; set; } = [];

    public ICollection<Photo> Photos { get; set; } = [];
}

public enum AppointmentStatus
{
    Pending,
    Confirmed,
    Cancelled,
    Completed
}

File: Frontend/src/app/types/appointment.ts
import { Photo, Service } from './service';

export type AppointmentStatus = 'Pending' | 'Confirmed' | 'Cancelled' | 'Completed';

export interface Appointment {
  id: number;
  appointmentDate: string;
  startTime: string;
  endTime: string;
  notes?: string | null;
  status: AppointmentStatus;
  clientName: string;
  clientEmail: string;
  clientPhone?: string | null;
  confirmationNotificationSent: boolean;
  confirmedAt?: string | null;
  created: string;
  services: Service[];
  photos: Photo[];
}

export interface AppointmentSettings {
  sendConfirmationNotifications: boolean;
  sendSms: boolean;
  sendEmail: boolean;
}

export interface AppointmentQueryParams {
  pageNumber?: number;
  pageSize?: number;
  search?: string;
  status?: AppointmentStatus | '' | null;
  fromDate?: string | null;
  toDate?: string | null;
  sortBy?: string;
  sortDirection?: 'asc' | 'desc';
}

export interface AppointmentFormValue {
  clientName: string;
  clientEmail: string;
  clientPhone?: string | null;
  appointmentDate: string;
  startTime: string;
  notes?: string | null;
}
